dataikuapi/dss/ml.py

```diff
@@ -0,0 +1,572 @@
+        Returns the original object, not a copy. Changes to the returned object are persisted to DSS by calling
+        :meth:`save_user_meta`
+
+        """
+        return self.details["userMeta"]
+
+    def save_user_meta(self):
+        um = self.details["userMeta"]
+
+        if self.mltask is not None:
+            self.mltask.client._perform_empty(
+                "PUT", "/projects/%s/models/lab/%s/%s/models/%s/user-meta" % (self.mltask.project_key,
+                    self.mltask.analysis_id, self.mltask.mltask_id, self.mltask_model_id), body = um)
+        else:
+            self.saved_model.client._perform_empty(
+                "PUT", "/projects/%s/savedmodels/%s/versions/%s/user-meta" % (self.saved_model.project_key,
+                    self.saved_model.sm_id, self.saved_model_version), body = um)
+
+class DSSTrainedPredictionModelDetails(DSSTrainedModelDetails):
+    """
+    Object to read details of a trained prediction model
+
+    Do not create this object directly, use :meth:`DSSMLTask.get_trained_model_details()` instead
+    """
+
+    def __init__(self, details, snippet, saved_model=None, saved_model_version=None, mltask=None, mltask_model_id=None):
+        DSSTrainedModelDetails.__init__(self, details, snippet, saved_model, saved_model_version, mltask, mltask_model_id)
+
+    def get_roc_curve_data(self):
+        roc = self.details.get("perf", {}).get("rocVizData",{})
+        if roc is None:
+            raise ValueError("This model does not have ROC visualization data")
+
+        return roc
+
+    def get_performance_metrics(self):
+        """
+        Returns all performance metrics for this model.
+
+        For binary classification model, this includes both "threshold-independent" metrics like AUC and
+        "threshold-dependent" metrics like precision. Threshold-dependent metrics are returned at the
+        threshold value that was found to be optimal during training.
+
+        To get access to the per-threshold values, use the following:
+
+        .. code-block:: python
+
+            # Returns a list of tested threshold values
+            details.get_performance()["perCutData"]["cut"]
+            # Returns a list of F1 scores at the tested threshold values
+            details.get_performance()["perCutData"]["f1"]
+            # Both lists have the same length
+
+        If K-Fold cross-test was used, most metrics will have a "std" variant, which is the standard deviation
+        accross the K cross-tested folds. For example, "auc" will be accompanied with "aucstd"
+
+        :returns: a dict of performance metrics values
+        :rtype: dict
+        """
+        import copy
+        clean_snippet = copy.deepcopy(self.snippet)
+        for x in ["gridsearchData", "trainDate", "topImportance", "backendType", "userMeta", "sessionDate", "trainInfo", "fullModelId", "gridLength", "algorithm", "sessionId"]:
```

Comment:
This looks difficult to remember to maintain, but I don't see a simple way around it :(.

Probably want to remove `topCoef` and `fullJobId` as well.